testsuite/tests/typing-modules/functors.ml

```diff
@@ -0,0 +1,1636 @@
+(* TEST
+  * expect
+*)
+
+
+
+module type a
+module type b
+module type c
+
+module type x = sig type x end
+module type y = sig type y end
+module type z = sig type z end
+
+
+module type empty = sig end
+
+module Empty = struct end
+module X: x = struct type x end
+module Y: y = struct type y end
+module Z: z = struct type z end
+module F(X:x)(Y:y)(Z:z) = struct end
+[%%expect {|
+module type a
+module type b
+module type c
+module type x = sig type x end
+module type y = sig type y end
+module type z = sig type z end
+module type empty = sig end
+module Empty : sig end
+module X : x
+module Y : y
+module Z : z
+module F : functor (X : x) (Y : y) (Z : z) -> sig end
+|}]
+
+
+module M = F(X)(Z)
+[%%expect {|
+Line 1, characters 11-18:
+1 | module M = F(X)(Z)
+               ^^^^^^^
+Error: The functor application is ill-typed.
+       These arguments:
+         X  Z
+       do not match these parameters:
+         functor (X : x) (Y : y) (Z : z) -> ...
+  1. Module X matches the expected module type x
```

Comment:
This was fixed, in a quite indirect way. The issue is that the box structure of the main message is not shared with submessages, it is thus impossible to recover the right alignment using the box structure.

However, using tabulation boxes works.

For future works, it might be interesting to look a bit more closely at the structure of error submessages however.